fdbserver/LogSystem.h

```diff
@@ -812,7 +815,10 @@ struct LogPushData : NonCopyable {
 	float getEmptyMessageRatio() const;
 
 	std::unordered_map<ptxn::TLogGroupID, std::shared_ptr<ptxn::ProxySubsequencedMessageSerializer>>*
-	    pGroupMessageBuilders;
+	    pGroupMessageBuilders = nullptr;
+
+	// Add TLog groups to initialize "pGroupMessageBuilders".
+	void addTLogGroups(const std::vector<TLogGroupRef>& groups, Version commitVersion);
```

Comment:
should this be used only for initialization? If so, consider removing this function and put the code directly in the constructor, to avoid misuse.

Reply:
Yes. However, non-PTXN code and proxy code don't use it. Only Resolver calls this function at `ResolverData::initGroupMessageBuilders` with a flag on. I think this way is cleaner.